shelley/chain-and-ledger/shelley-spec-ledger-test/test/Test/Shelley/Spec/Ledger/Rules/ClassifyTraces.hs

```diff
@@ -381,11 +338,12 @@ onlyValidChainSignalsAreGenerated =
     p :: Proxy C
     p = Proxy
 
-epochBoundariesInTrace :: [Block C] -> Int
-epochBoundariesInTrace bs =
-  length $
-    filter atEpochBoundary (blockSlot <$> bs)
+-- | Counts the epochs spanned by this trace
+epochsInTrace :: [Block C] -> Int
+epochsInTrace bs =
+  length . nub $
```

Comment:
Can we not count this much more easily by comparing the slots of the first and last blocks?

Reply:
yes you're right!